Code/Source/RobotImporter/RobotImporterWidget.cpp

```diff
@@ -30,20 +31,32 @@ namespace ROS2
     RobotImporterWidget::RobotImporterWidget(QWidget* parent)
         : QWidget(parent)
         , m_statusLabel("", this)
+        , m_selectFileButton(new QPushButton(QObject::tr("Load"), this))
+        , m_importerUpdateTimer(new QTimer(this))
+        , m_robotImporter(
+              [this](const AZStd::string& message)
+              {
+                  ReportInfo(message);
+              },
+              [this](const AZStd::string& message)
+              {
+                  ReportError(message);
+              })
     {
         setWindowTitle(QObject::tr("Robot definition file importer"));
         QVBoxLayout* mainLayout = new QVBoxLayout(this);
         mainLayout->setSpacing(20);
         QLabel* captionLabel = new QLabel(QObject::tr("Select a robot definition (URDF) file to import"), this);
         captionLabel->setWordWrap(true);
         mainLayout->addWidget(captionLabel);
-        QPushButton* selectFileButton = new QPushButton(QObject::tr("Load"), this);
-        mainLayout->addWidget(selectFileButton);
+        mainLayout->addWidget(m_selectFileButton);
         mainLayout->addWidget(&m_statusLabel);
         mainLayout->addStretch();
 
+        connect(m_importerUpdateTimer, &QTimer::timeout, this, QOverload<>::of(&RobotImporterWidget::ImporterTimerUpdate));
```

Comment:
Could we simply use a lambda here and do away with the member function? The name indicates it is only called from a timer.